buildpipeline/perf-pipeline.groovy

```diff
@@ -6,9 +6,374 @@
 
 //--------------------- Windows Functions ----------------------------//
 
-stage ('Basic') {
+def windowsBuild(String arch, String config, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String pgoBuildFlag = ((pgo == 'nopgo') ? '-nopgooptimize' : '-enforcepgo')
+    String baselineString = ""
+
+    // For baseline builds, checkout the merge's parent
+    if (isBaseline) {
+        baselineString = "-baseline"
+        bat "git checkout HEAD^^1"
+    }
+
+    bat "set __TestIntermediateDir=int&&.\\build.cmd -${config} -${arch} -skipbuildpackages ${pgoBuildFlag}"
+
+    // Stash build artifacts. Stash tests in an additional stash to be used by Linux test runs
+    stash name: "nt-${arch}-${pgo}${baselineString}-build-artifacts", includes: 'bin/**'
+    stash name: "nt-${arch}-${pgo}${baselineString}-test-artifacts", includes: 'bin/tests/**'
+}
+
+def windowsPerf(String arch, String config, String uploadString, String runType, String opt_level, String jit, String pgo, String scenario, boolean isBaseline) {
+    withCredentials([string(credentialsId: 'CoreCLR Perf BenchView Sas', variable: 'BV_UPLOAD_SAS_TOKEN')]) {
+        checkout scm
+        String baselineString = ""
+        if (isBaseline) {
+            baselineString = "-baseline"
+        }
+        dir ('.') {
+            unstash "nt-${arch}-${pgo}${baselineString}-build-artifacts"
```

Comment:
When you do this unstash does it end up with the folder name the same as the stash?  I ask because of max path considerations

Reply:
No, all the artifacts will unstash in the directory specified in dir in the structure they were stashed in.

Reply:
I thought that is how it worked, but wanted to make sure.